src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrVersionFileMerger.cs

```diff
@@ -0,0 +1,318 @@
+
+        List<DependencyUpdate> repoChanges = ComputeChanges(
+            previousRepoDependencies,
+            currentRepoDependencies);
+
+        List<DependencyUpdate> vmrChanges = ComputeChanges(
+            previousVmrDependencies,
+            currentVmrDependencies);
+
+        var mergedChanges = MergeDependencyUpdates(repoChanges, vmrChanges);
+
+        await ApplyVersionDetailsChangesAsync(targetRepo.Path, mergedChanges);
+
+        return mergedChanges;
+    }
+
+    private VersionFileChanges<JsonVersionProperty> MergeJsonChanges(
+        List<JsonVersionProperty> repoChanges,
+        List<JsonVersionProperty> vmrChanges)
+        => MergeVersionFileChanges(repoChanges, vmrChanges);
+
+    private VersionFileChanges<DependencyUpdate> MergeDependencyUpdates(
+        List<DependencyUpdate> repoChanges,
+        List<DependencyUpdate> vmrChanges)
+        => MergeVersionFileChanges(repoChanges, vmrChanges);
+
+    private VersionFileChanges<T> MergeVersionFileChanges<T>(
+        List<T> repoChanges,
+        List<T> vmrChanges) where T : IVersionFileProperty
+    {
+        var changedProperties = repoChanges
+            .Concat(vmrChanges)
+            .Select(c => c.Name)
+            .Distinct(StringComparer.OrdinalIgnoreCase);
+
+        var removals = new List<string>();
+        var additions = new Dictionary<string, T>();
+        var updates = new Dictionary<string, T>();
+        
+        foreach (var property in changedProperties)
+        {
+            var repoChange = repoChanges.FirstOrDefault(c => c.Name == property);
+            var vmrChange = vmrChanges.FirstOrDefault(c => c.Name == property);
+
+            var addedInRepo = repoChange != null && repoChange.IsAdded();
+            var addedInVmr = vmrChange != null && vmrChange.IsAdded();
+            var removedInRepo = repoChange != null && repoChange.IsRemoved();
+            var removedInVmr = vmrChange != null && vmrChange.IsRemoved();
+            var updatedInRepo = repoChange != null && repoChange.IsUpdated();
+            var updatedInVmr = vmrChange != null && vmrChange.IsUpdated();
+
+            if (removedInRepo)
+            {
+                if (addedInVmr)
+                {
+                    throw new ConflictingDependencyUpdateException(repoChange!, vmrChange!);
+                }
+                // we don't have to do anything since the property is removed in the repo
+                continue;
+            }
+
+            if (removedInVmr)
+            {
+                if (addedInRepo)
+                {
+                    throw new ConflictingDependencyUpdateException(repoChange!, vmrChange!);
+                }
+                removals.Add(property);
+                continue;
+            }
+
+            if (addedInRepo && addedInVmr)
+            {
+                additions[property] = (T)_mergerRegistry.SelectProperty(repoChange!, vmrChange!);
```

Comment:
Use the passed comparer method to do this simply
```suggestion
                additions[property] = propertySelector(repoChange!, vmrChange!);
```